/* Copyright 2023 The MediaPipe Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
==============================================================================*/

syntax = "proto3";

package mediapipe.tasks.vision.holistic_landmarker.proto;

import "mediapipe/framework/formats/classification.proto";
import "mediapipe/framework/formats/landmark.proto";

option java_package = "com.google.mediapipe.tasks.vision.holisticlandmarker";
option java_outer_classname = "HolisticResultProto";

message HolisticResult {
  mediapipe.NormalizedLandmarkList pose_landmarks = 1;
  mediapipe.LandmarkList pose_world_landmarks = 7;
  mediapipe.NormalizedLandmarkList left_hand_landmarks = 2;
  mediapipe.NormalizedLandmarkList right_hand_landmarks = 3;
  mediapipe.NormalizedLandmarkList face_landmarks = 4;
  mediapipe.ClassificationList face_blendshapes = 6;
  mediapipe.NormalizedLandmarkList auxiliary_landmarks = 5;
}
